src/Microsoft.AspNetCore.Proxy/ProxyService.cs

```diff
@@ -9,18 +9,28 @@ namespace Microsoft.AspNetCore.Proxy
 {
     public class ProxyService
     {
-        public ProxyService(IOptions<SharedProxyOptions> options)
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly HttpClient _httpClient;
+
+        public ProxyService(IOptions<SharedProxyOptions> options, IHttpClientFactory httpClientFactory)
         {
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
             Options = options.Value;
-            Client = new HttpClient(Options.MessageHandler ?? new HttpClientHandler { AllowAutoRedirect = false, UseCookies = false });
```

Comment:
@Tratcher how important is it for these options to be preserved. The HttpClientFactory doesn't have a great way of specifying HttpClientHandler options. See https://github.com/aspnet/HttpClientFactory/issues/110

Reply:
There options are critical. First one means the proxy want be able to properly forward 301 and 302 responses. Lack of the Second one hurts when using the legacy handler and ruins cookie forwarding to the server.